### PR TITLE
cask: audit if sparkle livecheck needs multipart version

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
I saw @miccal change this a bunch of times. Let's audit for it. Please let me know if I got the logic right.

### CODE REVIEW

**Comment by miccal** [comment]
Looks good, thanks @SMillerDev.

**Comment by SMillerDev** [comment]
Does it happen that the url contains 1 and 3? Or does sparkle only support 1 and 2?

**Comment by miccal** [comment]
> Does it happen that the url contains 1 and 3? Or does sparkle only support 1 and 2?

I think the basic `strategy :sparkle` cases are all of the form of either `version` or `version.csv.first,version.csv.second`.

The uses of `strategy :sparkle` in `do` blocks, however, may have more.

**Comment by SMillerDev** [comment]
I forgot about that possibility. I'll add a check for that too.

File: Library/Homebrew/cask/audit.rb [hunk 1]
```diff
@@ -474,6 +474,16 @@ def audit_download
       add_error "download not possible: #{e}"
     end
 
+    sig { void }
+    def audit_appcast_unneeded_long_version
+      return unless cask.livecheck.strategy == :sparkle
+      return unless cask.version.csv.second
+      return if cask.url.to_s.include? cask.version.csv.second
+      return if cask.url.to_s.include? cask.version.csv.third
```
Comment (reitermarkus) [inline Library/Homebrew/cask/audit.rb:482]: Does this actually work? I am thinking this will error when `#second` or `#third` return `nil`.

Comment (SMillerDev) [reply Library/Homebrew/cask/audit.rb:482]: For second that'll never happen since I check that above. For third it might

Comment (SMillerDev) [reply Library/Homebrew/cask/audit.rb:482]:
```suggestion
      return if cask.version.csv.third && cask.url.to_s.include? cask.version.csv.third
```

**Review by MikeMcQuaid — APPROVED** [review]

File: Library/Homebrew/cask/audit.rb [hunk 2]
```diff
@@ -474,6 +474,16 @@ def audit_download
       add_error "download not possible: #{e}"
     end
 
+    sig { void }
+    def audit_appcast_unneeded_long_version
+      return unless cask.livecheck.strategy == :sparkle
+      return unless cask.version.csv.second
+      return if cask.url.to_s.include? cask.version.csv.second
+      return if cask.version.csv.third.present? && cask.url.to_s.include? cask.version.csv.third
```
Comment (SMillerDev) [inline Library/Homebrew/cask/audit.rb:482]:
```suggestion
      return if cask.version.csv.third.present? && cask.url.to_s.include?(cask.version.csv.third)
```

**Pull request merged**